src/bug-filing/azure-boards/azure-boards-settings-form.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Dropdown, IDropdownOption, TextField } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { SettingsFormProps } from '../types/settings-form-props';
+import {
+    AzureBoardsBugFilingService,
+    AzureBoardsBugFilingSettings,
+    AzureBoardsIssueDetailField,
+    AzureBoardsIssueDetailLocationDropdownOption,
+} from './azure-boards-bug-filing-service';
+
+export const AzureBoardsSettingsForm = NamedSFC<SettingsFormProps<AzureBoardsBugFilingSettings>>('AzureBoardsSettingsForm', props => {
+    const defaultKey = 'reproSteps';
+    const options: AzureBoardsIssueDetailLocationDropdownOption[] = [
+        {
+            key: defaultKey,
+            text: 'Repro steps (default)',
+        },
+        {
+            key: 'description',
+            text: 'Description',
+        },
+    ];
+
+    const onProjectURLChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+        const propertyName: keyof AzureBoardsBugFilingSettings = 'projectURL';
+        props.deps.userConfigMessageCreator.setBugServiceProperty(AzureBoardsBugFilingService.key, propertyName, newValue);
+    };
+
+    const onIssueDetailLocationChange = (event: React.FormEvent<HTMLElement>, newValue?: IDropdownOption) => {
+        const propertyName: keyof AzureBoardsBugFilingSettings = 'issueDetailsField';
+        props.deps.userConfigMessageCreator.setBugServiceProperty(
+            AzureBoardsBugFilingService.key,
+            propertyName,
+            newValue.key as AzureBoardsIssueDetailField,
+        );
+    };
+
+    return (
+        <>
+            <TextField
+                className="issue-setting"
+                label="Enter the desired Azure Boards project link:"
+                value={props.settings ? props.settings.projectURL : null}
```

Comment:
`null` isn't valid here - results in a console error (below).
```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
    in input (created by TextFieldBase)
    in div (created by TextFieldBase)
    in div (created by TextFieldBase)
    in div (created by TextFieldBase)
    in TextFieldBase (created by Context.Consumer)
    in StyledTextFieldBase (created by AzureBoardsSettingsForm)
    in AzureBoardsSettingsForm (created by BugFilingSettingsContainer)
    in BugFilingSettingsContainer (created by IssueFilingDialog)
```

Reply:
Where are you getting this?

Reply:
I tried adding `AzureBoardsBugFilingService` to `BugFilingServiceProviderImpl` so I could see the UX. Then I cleaned settings, did a FastPass with failures, clicked File Issue in the failure details pane, selected Azure Boards, and saw this error on the console.

Reply:
Sure... it's fixed now.